File: frontend/src/shared/api/dtos/properties/property-list-item.dto.ts
import type { PropertyType} from "@/shared/types/properties.ts";
import type { PropertyStatus } from "@/shared/types/properties.ts";

export type PropertyListItemDto = {
  id: string;
  title: string;
  city: string;
  price: number;
  type: PropertyType;
  status: PropertyStatus;
  mainImageUrl?: string;
  createdAt?: string;
  // TODO: add brokerID(?) and updatedAt at backend and here
}

File: backend/src/Application/Features/Properties/List/PropertyListItemDto.cs
using RealEstate.Domain.Enums.Properties;

namespace RealEstate.Application.Features.Properties.List;

public sealed record PropertyListItemDto(
    int Id,
    int? AgencyId,
    int? BrokerId,
    string Title,
    string City,
    decimal Price,
    string? MainImageUrl,
    PropertyType Type,
    PropertyStatus Status,
    DateTime CreatedAt
);
